bftengine/src/bftengine/messages/ClientRequestMsg.cpp

```diff
@@ -102,11 +102,19 @@ void ClientRequestMsg::validateImp(const ReplicasInfo& repInfo) const {
   const auto* header = msgBody();
   PrincipalId clientId = header->idOfClientProxy;
   if ((header->flags & RECONFIG_FLAG) == 0) ConcordAssert(this->senderId() != repInfo.myId());
+
```

Comment:
@agrkushvm , I think execution might potentially crash already in line 103, you have to place you check before accessing the header fields.
To safetly access any field in header you must check first that
`(msgSize >= sizeof(ClientRequestMsgHeader)`. 

In addition, line 177 is doing a similar check, but potentially too late.
After validating the 1st check and being able to access the header, I would move the check (msgSize < minMsgSize) , i would move this check to the top of the function.
So you need to have 2 checks at the beggining of the function:

1. Before accessing any field in messge body, check that (msgSize >= sizeof(ClientRequestMsgHeader) + spanContextSize())
2. Immidiately after check that  (msgSize < minMsgSize)

Reply:
ok